### PR TITLE
route to get all employees

### PR DESCRIPTION
**Pull request by abibou1**
There are two main changes:
-I added Visit as a new Mongoose schema. 
-The route to get all employees:
removePassword

### CODE REVIEW

File: server/src/api/v1/routes/visitors.js [hunk 1]
```diff
@@ -1,9 +1,44 @@
...
 
 const router = Router();
 
-router.post('/checkIn', (req, res) => {
-  res.send('Checked In');
+router.get('/admin/users', (req, res) => {
+  User.find()
+    .then((employees) => {
+      res.status(200).json({ employees });
+    })
```
Comment (mazma1) [inline server/src/api/v1/routes/visitors.js:11]: I've seen your approach of excluding the fields you don't want returned to the client (`password` and `_v`). You could have also done something like this according to the Stack Overflow responses [here](https://stackoverflow.com/a/32630779/13259158) to achieve something:

```
User.find({}, '-password -_v')
   .then((employees) => {
    res.status(200).json({ employees });
})
```
You can also give it a try to see if it works (you don't need to change what you've done, just wanted to share for learning purpose)

File: server/src/api/v1/routes/visitors.js [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
+    .catch((error) => res.status(500).json(error));
 });
 
+/*
```
[the remark this answers is not in the document]
Comment (abibou1) [reply server/src/api/v1/routes/visitors.js]: I wanted to keep it here as I would need it for that route. I removed it now.

File: server/src/api/v1/routes/visitors.js (hunk 1 above)
Comment (abibou1) [reply server/src/api/v1/routes/visitors.js]: I like this method of excluding as opposed to a deletion.  I keep the other one in the comment for future reference. You can merge this branch if everything is fine.

**Comment by mazma1** [comment]
One more thing, did you add the condition that ensures that only an admin can use this route and returns appropriate error message if the user is not an admin? I think you missed that part.

**Comment by abibou1** [comment]
### **This is when a user is not an admin and tries to access data**
nonAdmin


**### Admin:**
I created a user john1@gmail.com with isAdmin=true in the database. So that user, if authencated, can access data.
admin

**Comment by mazma1** [comment]
Kindly check, the UI of the review app is not loading. Link: https://myguesty2-pi-ft-routes--fxpeq3.herokuapp.com/

File: server/src/api/v1/middleware/authenticatedAdmin.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import jwt from 'jsonwebtoken';
+
+module.exports = (req, res, next) => {
+  try {
+    const token = req.headers.authorization.split(' ')[1];
```
Comment (mazma1) [inline server/src/api/v1/middleware/authenticatedAdmin.js:5]: I also get this error which appears to be coming from here:

Screenshot 2020-11-05 at 07 43 42


So you need to confirm that there is a token in the header first and return the appropriate error if there is none.

File: server/src/api/v1/middleware/authenticatedAdmin.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    const userEmail = decodedToken.email;
+    if (req.body.email && req.body.email !== userEmail) {
+      throw new Error('Invalid user email');
+    } else if (decodedToken.isAdmin === false) {
+      throw new Error('Must be an admin');
+    } else {
+      next();
+    }
+  } catch (err) {
+    res.status(401).json({ error: err.message });
```
Comment (mazma1) [inline server/src/api/v1/middleware/authenticatedAdmin.js:16]: It's also important to note that when a token is missing from the header (user is not logged in), the correct status code is `401`. However, when the token is present (user is successfully authenticated) but is not an admin, the correct status code should be `403`.

Found a concise explanation [here](https://www.g33kzone.com/2018/05/http-status-codes-401-vs-403.html).

File: server/src/api/v1/routes/visitors.js [hunk 5]
```diff
@@ -1,9 +1,15 @@
 import { Router } from 'express';
+import User from '../../../models/User';
+import authenticatedAdmin from '../middleware/authenticatedAdmin';
 
 const router = Router();
 
-router.post('/checkIn', (req, res) => {
-  res.send('Checked In');
+router.get('/admin/users', authenticatedAdmin, (req, res) => {
```
Comment (mazma1) [inline server/src/api/v1/routes/visitors.js:7]: Great use of a middleware here! 👍🏽

**Comment by abibou1** [comment]
I made the changes. Also, I removed the email comparison(req.body.email === decodedToken.email). It is a GET route, so I don't expect to receive any information. I hope that is fine to do.

**Comment by mazma1** [comment]
Yes, removing the email comparison is correct

**Pull request merged**